MacBry-Maciej-Bryja/SimpleTodoApp1/src/main/java/com/mac/bry/simpleTodo/DAO/UserDAO.java

```diff
@@ -0,0 +1,198 @@
+package com.mac.bry.simpleTodo.DAO;
+
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+import com.mac.bry.simpleTodo.DAO.API.DAOUserAPI;
+import com.mac.bry.simpleTodo.entity.Task;
+import com.mac.bry.simpleTodo.entity.User;
+
+public class UserDAO implements DAOUserAPI {
+	
+	private SessionFactory factory;
+	
+	public UserDAO() {
+		super();
+		this.factory = new Configuration()
+				.configure("hibernate.cfg.xml")
+				.addAnnotatedClass(User.class)
+				.addAnnotatedClass(Task.class)
+				.buildSessionFactory();
+	}
+	
+	@Override
+	public void addUser(User user) {
+		List<User> userList;
+		Session session = factory.getCurrentSession();
+		session.beginTransaction();
+		userList = session.createQuery("from User user where user.login='" + user.getLogin() + "'").getResultList();
+		if(userList.isEmpty()) {
```

Comment:
Zbędne - wystarczy na kolumnie login zapiąć constrainta 'unique' i w przypadku, gdy dojdzie do duplikacji to hibernate rzuci wyjątkiem,

Reply:
Zrobione!